exporter/datadogexporter/internal/model/translator/metrics_translator.go

```diff
@@ -396,28 +405,25 @@ func (t *Translator) MapMetrics(ctx context.Context, md pmetric.Metrics, consume
 
 		// Fetch tags from attributes.
 		attributeTags := attributes.TagsFromAttributes(rm.Resource().Attributes())
-
-		host, ok := attributes.HostnameFromAttributes(rm.Resource().Attributes(), t.cfg.previewHostnameFromAttributes)
+		src, ok := attributes.SourceFromAttributes(rm.Resource().Attributes(), t.cfg.previewHostnameFromAttributes)
 		if !ok {
 			var err error
-			host, err = t.cfg.fallbackHostnameProvider.Hostname(context.Background())
+			src, err = t.cfg.fallbackSourceProvider.Source(context.Background())
 			if err != nil {
-				return fmt.Errorf("failed to get fallback host: %w", err)
+				return fmt.Errorf("failed to get fallback source: %w", err)
 			}
 		}
 
-		if host != "" {
-			// Track hosts if the consumer is a HostConsumer.
+		var host string
+		switch src.Kind {
+		case source.HostnameKind:
+			host = src.Identifier
 			if c, ok := consumer.(HostConsumer); ok {
-				c.ConsumeHost(host)
+				c.ConsumeHost(src.Identifier)
```

Comment:
Since we define `host` a few lines above, can we use it here?

Reply:
Done!